File: src/0001-Two-Sum/two_sum.go
package main

import (
	"fmt"
)

func twoSum(nums []int, target int) []int {
	nums_hash := map[int]int{}
	for i, num := range nums {
		dif := target - num
		if v, ok := nums_hash[dif]; ok {
			return []int{v, i}
		} 
		nums_hash[nums[i]] = i
	}
	return nil
}

func main() {
	nums := []int{1, 2, 3}
	target := 3
	fmt.Println(twoSum(nums, target))
}

File: src/0001-Two-Sum/two_sum.js
var twoSum = function (nums, target) {
    for (let i = 0; i < nums.length; ++i) {
        for (let j = i + 1; j < nums.length; ++j) {
            if (nums[i] + nums[j] == target) {
                return [i, j];
            }
        }
    }
};
